File: java/v1/src/test/java/com/api/v1/payment/AllPaymentsRetrievalTest.java
package com.api.v1.payment;

import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.web.reactive.server.WebTestClient;

@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
public class AllPaymentsRetrievalTest {

    @Autowired
    private WebTestClient webTestClient;

    @Test
    void testIfPaymentIsNotEmpty() {
        webTestClient
                .get()
                .uri("api/v1/payments")
                .exchange()
                .expectStatus()
                .is2xxSuccessful();
    }

}

File: v2/src/test/kotlin/com/api/v2/payment/AllPaymentsRetrievalTest.kt
package com.api.v2.payment

import org.junit.jupiter.api.MethodOrderer
import org.junit.jupiter.api.Order
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.TestMethodOrder
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.web.reactive.server.WebTestClient

@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
private class AllPaymentRetrievalTest {

    @Autowired
    lateinit var webTestClient: WebTestClient

    @Test
    fun testSuccessfulAllPaymentsRetrieval() {
        webTestClient
            .get()
            .uri("api/v1/payments")
            .exchange()
            .expectStatus()
            .is2xxSuccessful()
    }

}